src/main/java/org/sonarlint/intellij/inspection/SonarLintGlobalInspection.java

```diff
@@ -141,20 +144,61 @@ public void visitFile(PsiFile psiFile) {
   }
 
   ProblemDescriptor getProblemDescriptor(Issue issue, PsiFile psiFile, @NotNull InspectionManager manager, boolean isLocal) {
-    if (issue.getLine() != null) {
-      return manager.createProblemDescriptor(psiFile,
-        InspectionUtils.getLineRange(psiFile, issue.getLine()),
-        InspectionUtils.getProblemMessage(issue, isLocal),
-        problemHighlightType(issue, isLocal),
-        false);
+    Integer startLine = issue.getStartLine();
+    if (startLine != null) {
+      Integer startLineOffset = issue.getStartLineOffset();
+      if (startLineOffset != null) {
+        Integer endLine = issue.getEndLine();
+        Integer endLineOffset = issue.getEndLineOffset();
+        int ijStartLine = startLine - 1;
+        int ijEndLine = endLine - 1;
+        final Document document = PsiDocumentManager.getInstance(psiFile.getProject()).getDocument(psiFile);
+        PsiElement startElement;
+        PsiElement endElement;
+        if (document != null) {
+          int lineStartOffset = document.getLineStartOffset(ijStartLine);
+          final int startOffset = lineStartOffset + startLineOffset;
+          final int endOffset = (ijEndLine == ijStartLine) ? lineStartOffset + endLineOffset : document.getLineStartOffset(ijEndLine) + endLineOffset;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS864)